### DISCUSSION TITLE
Vue `useHead` composable instead of `<Head>` component ?

### DISCUSSION THREAD

**Discussion by mrleblanc101** [inertiajs/inertia#1810] [Ideas]
Wouldn't that be more clean ? 
I hate to place a `Head` in my components that is rendered elsewhere in the DOM.
I don't feel like it's intuitive or a good practive.
I think a `useHead` composable like Nuxt would be cleaner. https://nuxt.com/docs/api/composables/use-head
Nuxt `useHead` is powered by `unhead` which is framework agnostic so it could be implemented in Inertia: https://unhead.unjs.io/

**Comment by SergkeiM** [comment]
Hi @mrleblanc101 
Totally agree, in my custom Inertia build, I've repalced inertia header manager with `unhead` works like a charm and has more functionalities.

**Comment by Borderliner** [comment]
@SergkeiM How do you do this?

**Comment by SergkeiM** [comment]
Hi @Borderliner,

InertiaJS is under MIT license, so I took the source code and did the changes:

inside `createInertiaApp`

```js
import { createHead } from '@unhead/vue'
import { renderSSRHead } from '@unhead/ssr'
....
createInertiaApp({...}
....
const head = createHead()

const vueApp = await resolveComponent(initialPage.component).then((initialComponent) => {
        return setup({
            el,
            App,
            head, // return created head
            props: {
                initialPage,
                initialComponent,
                resolveComponent
            }
        })
    })
...

// If inside ssr:
const payload = await renderSSRHead(head)

return {
      head: payload.headTags,
      bodyTags: payload.bodyTags,
      bodyTagsOpen: payload.bodyTagsOpen,
      htmlAttrs: payload.htmlAttrs,
      bodyAttrs: payload.bodyAttrs,
      body
  }
  
 //Inside your setup
 
 setup: ({ el, App, props, head }) => createSSRApp({
        render: () => h(App, props)
    })
    .use(head).mount(el);
```

**Comment by Borderliner** [comment]
@SergkeiM Would appreciate seeing your fork or a diff.
